File: arrays/add-one-to-number.cpp
vector<int> Solution::plusOne(vector<int> &A) {
    int n = A.size();
    vector<int> v,ans;
    int c=1, k, r;

    for(int i=n-1; i>=0; i--){
        k=A[i]+c;
        c=k/10;
        r=k%10;
        v.push_back(r);
    }
    v.push_back(c);
    while(!v.back()) v.pop_back();
    
    reverse(v.begin(), v.end());
    return v;
}

File: arrays/add-one-to-number.py
class Solution:
    def plusOne(self, A):
        c = 1
        while len(A) and A[0]==0: A.pop(0)
        for i in range(len(A)-1, -1, -1):
            add = A[i] + c
            A[i] = int(add%10)
            c = int(add/10)
        if c==1: A.insert(0, c)
        return A
